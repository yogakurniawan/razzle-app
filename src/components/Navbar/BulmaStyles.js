import styled, { css } from 'styled-components'

const baseNavbar = css`
  @media screen and (min-width: 1088px) {
    align-items: stretch;
    display: flex;
  }
`

export const Navbar = styled.nav`
  background-color: #fff;
  color: #0a0a0a;
  min-height: 3.25rem;
  position: relative;
  z-index: 30;
  box-shadow: 0 2px 0 0 #f5f5f5;
  ${baseNavbar}
  @media screen and (min-width: 1088px) {
    min-height: 3.25rem;
  }
`

export const NavbarMenu = styled.div`
  ${baseNavbar}
  @media screen and (min-width: 1088px) {
    flex-grow: 1;
    flex-shrink: 0;
  }
  @media screen and (max-width: 1087px) {
    background-color: #fff;
    box-shadow: 0 8px 16px rgba(10,10,10,.1);
    padding: .5rem 0;
  }
`

export const NavbarStart = styled.div`
  ${baseNavbar}
  @media screen and (min-width: 1088px) {
    justify-content: flex-start;
    margin-right: auto;
  }
`

export const NavbarBrand = styled.div`
  align-items: stretch;
  display: flex;
  flex-shrink: 0;
  min-height: 3.25rem;
`

export const NavbarBurger = styled.div`
  cursor: pointer;
  display: block;
  height: 3.25rem;
  position: relative;
  width: 3.25rem;
  margin-left: auto;
  @media screen and (min-width: 1088px) {
    display: none;
  }
  &:hover {
    background-color: rgba(0,0,0,.05);
  }
  > span {
    background-color: currentColor;
    display: block;
    height: 1px;
    left: calc(50% - 8px);
    position: absolute;
    transform-origin: center;
    transition-duration: 86ms;
    transition-property: background-color,opacity,transform;
    transition-timing-function: ease-out;
    width: 16px;
    &:nth-child(1) {
      top: calc(50% - 6px);
    }
    &:nth-child(2) {
      top: calc(50% - 1px);
    }
    &:nth-child(3) {
      top: calc(50% + 4px);
    }
  }
  ${({ state }) => state === 'active' &&
    css`
      span:nth-child(1) {
        transform: translateY(5px) rotate(45deg);
      }
    `
  }
`

export const NavbarItem = styled.a`
  color: #0a0a0a;
  cursor: pointer;
  text-decoration: none;
  flex-grow: 0;
  flex-shrink: 0;
  line-height: 1.5;
  padding: .5rem .75rem;
  position: relative;
  @media screen and (min-width: 1088px) {
    display: flex;
    align-items: center;
  }
  &:hover {
    background-color: #f2f2f2;
  }
`