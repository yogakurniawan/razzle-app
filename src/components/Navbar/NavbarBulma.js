import React from 'react'
import {
  Navbar,
  NavbarMenu,
  NavbarItem,
  NavbarStart,
  NavbarBrand,
  NavbarBurger
} from './BulmaStyles'

export default function NavbarBulma() {
  return (
    <Navbar>
      <NavbarBrand>
        <NavbarBurger state="inactive">
          <span></span>
          <span></span>
          <span></span>
        </NavbarBurger>
      </NavbarBrand>
      <NavbarMenu>
        <NavbarStart>
          <NavbarItem href="">
            Home
          </NavbarItem>
        </NavbarStart>
      </NavbarMenu>
    </Navbar>
  )
}