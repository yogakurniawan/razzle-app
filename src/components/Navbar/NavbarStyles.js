import styled, { css } from 'styled-components'
import { Icon } from '../Icon'

export const NavbarFixed = styled.header`
  border-bottom: 1px solid #e4e4e4;
  z-index: 5;
`

export const NavbarNavigation = styled.ul`
  @media screen and (min-width: 768px) {
    flex-flow: row;
    justify-content: flex-end;
  }
  display: flex;
  flex-direction: column;
  list-style: none;
  padding-left: 0;
  padding-right: 20px;
  margin: 0;
`

export const NavbarHeader = styled.li`
  @media screen and (min-width: 768px) {
    margin-right: auto;
  }
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  align-items: center;
  padding-right: 10px;
  width: 100%;
`

export const BaseAStyle = css`
  text-decoration: none;
  color: inherit;
  cursor: pointer;
`

export const ArrowDown = css`
  ${Icon}
  content: '\f078';
`

export const ArrowDownWrapper = styled.label`
  @media screen and (min-width: 768px) {
    display: none;
  }
  display: inline-block;
  font-size: 9px;
  margin-left: 8px;
  vertical-align: middle;
  transition-property: -ms-transform,-webkit-transform,transform;
  transition-duration: 250ms;
  transition-timing-function: ease-in-out;
  span {
    &:before {
      ${ArrowDown}      
    }
  }
`

export const NavbarMenu = styled.div`
  @media screen and (min-width: 768px) {
    overflow: visible;
  }
  background-color: #fff;
  color: #484848;
  width: 100%;
  transition: 250ms ease-in-out;

  ${props => !props.expand &&
    css`
      @media screen and (min-width: 768px) {
        overflow: visible;
      }
      overflow: hidden;
    `}

  ${props => props.expand &&
    css`
      position: absolute;
      transition: height 250ms ease-in-out;
      height: 100vh;
      overflow: hidden;
      ${ArrowDownWrapper} {
        transform: rotate(180deg);
      }
    `}
`

export const NavbarBrand = styled.div`
  @media screen and (max-width: 320px) {
    padding: 10px;
  }
  cursor: pointer;
  ${ArrowDownWrapper} {
    cursor: pointer;
  }
  margin-right: 10px;
  padding: 6px 10px 6px 10px;
  display: table;
`

export const NavbarItem = styled.li`
  @media screen and (min-width: 768px) {
      border-top: 0;
      font-size: 1em;
  }
  font-size: 19px;
  a {
    ${BaseAStyle}
    padding: 18px 15px 18px 15px;
    display: block;
  }
`

export const ButtonMenu = styled.button`
  appearance: none;
  background: transparent;
  border: none;
  color: inherit;
  display: inline-block;
  text-decoration: none;
  margin: 0px;
  position: relative;
  white-space: nowrap;
`

export const Item = styled.div`
  @media (min-width: 768px) {
    font-weight: normal;
    a {
      position: relative;
      border-bottom: ${props => (props.active ? '2px solid #484848 !important;' : '2px solid #fff !important;')}
      &:hover {
        &:after {
          width: 100%;
        }
      }
      ${props => !props.active && css`
        &:after, &:before {
          transition: all .2s;
        }
        &:after {
          position: absolute;
          bottom: 0;
          left: 0;
          right: 0;
          margin: auto;
          width: 0%;
          content: '.';
          color: transparent;
          background: #484848;
          height: 2px;
        }
      `}
    }
  }
  padding: 8px 0;
  font-weight: 300;
  line-height: 1;
  a {
    border-bottom: 2px solid #fff;
  }
`

const IconCss = css`
  height: 1em;
  width: 1em;
  display: block;
  fill: currentColor;
`

export const SearchIconWrapper = styled.div`
  float: left;
  @media screen and (min-width: 768px) {
    padding-left: 12px;
    padding-top: 12px;
  }
  padding-left: 0;
  padding-top: 16px;
`

export const SearchInputWrapper = styled.div`
overflow: hidden;
position: relative;
`

export const SearchInput = styled.input`
  font-family: Circular,"Helvetica Neue",Helvetica,Arial,sans-serif;
  font-size: 19px;
  line-height: 24px;
  color: #484848;
  background-color: transparent;
  border: 0;
  padding: 12px;
  width: 100%;
  font-weight: normal;
  text-overflow: ellipsis;
  &:focus {
    outline: none;
  }
`

export const SearchBox = styled.div`
  width: 100%;
  line-height: 24px;
  color: #484848;
  background-color: #ffffff;
  border-radius: 4px !important;
  border: 1px solid #DBDBDB;
  box-shadow: 0 2px 4px rgba(0, 0, 0, 0.1);
  padding: 0 8px;
  height: 50px;
  margin-right: auto;
  white-space: nowrap;
  position: relative;
  @media screen and (min-width: 1168px) {
    width: 560px;
  }
  @media screen and (min-width: 1075px) {
    width: 460px;
  }
  @media screen and (min-width: 1040px) {
    width: 500px;
  }
`

export const ContainerFluid = styled.div`
  width: 100%;
  top: 0;
  left: 0;
  position: fixed;
  z-index: 10;
`
