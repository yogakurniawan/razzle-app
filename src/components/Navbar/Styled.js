import styled, { css } from 'styled-components'
import { Link } from 'react-router-dom'

const baseNavbar = css`
  @media screen and (min-width: 1088px) {
    align-items: stretch;
    display: flex;
  }
`

const baseLink = css`
  cursor: pointer;
  text-decoration: none;
  color: #777;
  line-height: 1.5;
`

const baseNavbarItem = css`
  ${({ theme, state }) =>
    css`
      ${baseLink}
      font-size: 1rem;
      display: block;
      flex-grow: 0;
      flex-shrink: 0;
      padding: .5rem .75rem;
      position: relative;
      @media screen and (min-width: 1088px) {
        display: flex;
        align-items: center;
      }
      &:hover {
        background-color: #f7f7f7;
        color: ${theme.primary};
      }
      ${state === 'has-dropdown' &&
        css`
          padding: 0;
          @media screen and (min-width: 1088px) {
            align-items: stretch;
          }
      `}
  `}
`

export const NavbarLabel = styled.span`
  ${({ theme, state }) =>
    css`
      ${baseLink}
      display: block;
      padding: .5rem .75rem;
      position: relative;
      padding-right: 2.5em;
      &:hover {
        background-color: #f7f7f7;
        color: ${theme.primary};
      }
      @media screen and (min-width: 1088px) {
        align-items: center;
        display: flex;
      }
      &::after {
        margin-top: -.375em;
        right: 1.125em;
        border: 3px solid ${theme.primary};
        border-radius: 2px;
        border-right: 0;
        border-top: 0;
        content: " ";
        display: block;
        height: .625em;
        pointer-events: none;
        position: absolute;
        top: 50%;
        transform: rotate(-45deg);
        transform-origin: center;
        width: .625em;
      }
  `}
`

export const Navbar = styled.nav`
  background-color: #fff;
  color: #0a0a0a;
  min-height: 3.25rem;
  position: relative;
  z-index: 30;
  box-shadow: 0 2px 0 0 #f5f5f5;
  ${baseNavbar}
  @media screen and (min-width: 1088px) {
    min-height: 3.25rem;
    padding: 1rem 2rem;
  }
`

export const NavbarMenu = styled.div`
  ${baseNavbar}
  display: none;
  @media screen and (min-width: 1088px) {
    flex-grow: 1;
    flex-shrink: 0;
  }
  @media screen and (max-width: 1087px) {
    background-color: #fff;
    box-shadow: 0 8px 16px rgba(10,10,10,.1);
    padding: .5rem 0;
    ${({ state }) => state === 'active' &&
      css`
        display: block;
      `
    }
  }
`

export const NavbarStart = styled.div`
  ${baseNavbar}
  @media screen and (min-width: 1088px) {
    justify-content: flex-start;
    margin-right: auto;
  }
`

export const NavbarEnd = styled.div`
  ${baseNavbar}
  @media screen and (min-width: 1088px) {
    justify-content: flex-end;
    margin-left: auto;
  }
`

export const NavbarBrand = styled.div`
  align-items: stretch;
  display: flex;
  flex-shrink: 0;
  min-height: 3.25rem;
`

export const NavbarBurger = styled.div`
  cursor: pointer;
  display: block;
  height: 3.25rem;
  position: relative;
  width: 3.25rem;
  margin-left: auto;
  @media screen and (min-width: 1088px) {
    display: none;
  }
  &:hover {
    background-color: rgba(0,0,0,.05);
  }
  > span {
    background-color: currentColor;
    display: block;
    height: 1px;
    left: calc(50% - 8px);
    position: absolute;
    transform-origin: center;
    transition-duration: 86ms;
    transition-property: background-color,opacity,transform;
    transition-timing-function: ease-out;
    width: 16px;
    &:nth-child(1) {
      top: calc(50% - 6px);
    }
    &:nth-child(2) {
      top: calc(50% - 1px);
    }
    &:nth-child(3) {
      top: calc(50% + 4px);
    }
  }
  ${({ state }) => state === 'active' &&
    css`
      span:nth-child(1) {
        transform: translateY(5px) rotate(45deg);
      }
      span:nth-child(2) {
        opacity: 0;
      }
      span:nth-child(3) {
        transform: translateY(-5px) rotate(-45deg);
      }
    `
  }
`

export const NavbarDropDown = styled.div`
  ${({ theme, position }) =>
    css`
      ${NavbarItemLink} {
        font-size: .875rem;
        padding-left: 1.5rem;
        padding-right: 1.5rem;
      }
      padding-bottom: .5rem;
      padding-top: .5rem;
      @media screen and (min-width: 1088px) {
        background-color: ${theme.white};
        border-bottom-left-radius: 6px;
        border-bottom-right-radius: 6px;
        border-top: 2px solid #dbdbdb;
        box-shadow: 0 8px 8px rgba(10,10,10,.1);
        display: none;
        left: 0;
        min-width: 10rem;
        position: absolute;
        top: 100%;
        z-index: 20;
        ${position === 'right' && css`
          left: auto;
          right: 0;
        `}
      }
  `}
`

export const NavbarItem = styled.div`
  ${baseNavbarItem}
`

export const NavbarItemLink = styled(Link)`
  ${baseNavbarItem}
`

export const NavbarItemDiv = styled.div`
  ${baseNavbarItem}
  &:hover {
    ${NavbarDropDown} {
      display: block;
    }
    background-color: initial;
  }
`