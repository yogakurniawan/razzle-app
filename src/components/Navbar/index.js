import React, { Component } from 'react';
import { User2 } from 'components/Icon'
import {
  Navbar,
  NavbarMenu,
  NavbarItem,
  NavbarItemLink,
  NavbarItemDiv,
  NavbarStart,
  NavbarBrand,
  NavbarBurger,
  NavbarLabel,
  NavbarDropDown,
  NavbarEnd
} from './Styled'

export default class NavbarBulma extends Component {
  state = {
    active: false
  }

  toggleState = () => {
    this.setState({
      active: !this.state.active
    })
  }

  render() {
    const { active } = this.state
    const { logout } = this.props
    return (
      <Navbar>
        <NavbarBrand>
          <NavbarBurger data-testid="navbar-burger" onClick={this.toggleState} state={active ? 'active' : 'inactive'}>
            <span></span>
            <span></span>
            <span></span>
          </NavbarBurger>
        </NavbarBrand>
        <NavbarMenu state={active ? 'active' : 'inactive'}>
          <NavbarStart>
            <NavbarItemLink to="/">
              Home
            </NavbarItemLink>
            <NavbarItemLink to="/appointments">
              Appointments
            </NavbarItemLink>
            <NavbarItemLink to="/waitingroom">
              Waiting Room
            </NavbarItemLink>
            <NavbarItemLink to="/patients">
              Patients
            </NavbarItemLink>
            <NavbarItemDiv state="has-dropdown">
              <NavbarLabel>
                Accounts
              </NavbarLabel>
              <NavbarDropDown>
                <NavbarItemLink to="/sales">
                  Sales
                </NavbarItemLink>
                <NavbarItemLink to="/expenses">
                  Expenses
                </NavbarItemLink>
                <NavbarItemLink to="/distributions">
                  Distributions
                </NavbarItemLink>
                <NavbarItemLink to="/journal">
                  Journal
                </NavbarItemLink>
                <NavbarItemLink to="/reconciliation">
                  Reconciliation
                </NavbarItemLink>
                <NavbarItemLink to="/acir">
                  AIR - Claiming
                </NavbarItemLink>
                <NavbarItemLink to="/claiming">
                  Claiming
                </NavbarItemLink>
              </NavbarDropDown>
            </NavbarItemDiv>
            <NavbarItemDiv state="has-dropdown">
              <NavbarLabel>
                More
              </NavbarLabel>
              <NavbarDropDown>
                <NavbarItemLink to="/resources">
                  Resources
                </NavbarItemLink>
                <NavbarItemLink to="/settings">
                  Settings
                </NavbarItemLink>
                <NavbarItemLink to="/reporting">
                  Reporting
                </NavbarItemLink>
                <NavbarItemLink to="/audit">
                  Audit
                </NavbarItemLink>
                <NavbarItemLink to="/utilities">
                  Utilities
                </NavbarItemLink>
              </NavbarDropDown>
            </NavbarItemDiv>
          </NavbarStart>
          <NavbarEnd>
            <NavbarItemDiv state="has-dropdown">
              <NavbarLabel>
                <User2 />
              </NavbarLabel>
              <NavbarDropDown position="right">
                <NavbarItem onClick={logout}>
                  Logout
                </NavbarItem>
              </NavbarDropDown>
            </NavbarItemDiv>
          </NavbarEnd>
        </NavbarMenu>
      </Navbar>
    )
  }
}