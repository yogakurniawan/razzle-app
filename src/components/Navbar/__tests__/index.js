import React from 'react'
import { Router } from 'react-router-dom'
import { createMemoryHistory } from 'history'
import { render, fireEvent, cleanup } from 'react-testing-library'
import Navbar from '../index'

function renderWithRouter(
  ui,
  { route = '/', history = createMemoryHistory({ initialEntries: [route] }) } = {},
) {
  return {
    ...render(<Router history={history}>{ui}</Router>),
    // adding `history` to the returned utilities to allow us
    // to reference it in our tests (just try to avoid using
    // this to test implementation details).
    history,
  }
}

test('Navbar burger has state inactive', () => {
  const logout = jest.fn()
  const { container } = renderWithRouter(<Navbar logout={logout} />)
  expect(container).toMatchSnapshot()
})