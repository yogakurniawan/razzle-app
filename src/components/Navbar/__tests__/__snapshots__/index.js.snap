// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar burger has state inactive 1`] = `
<div>
  <nav
    class="Styled__Navbar-fbosc2-1 gwKhqW"
  >
    <div
      class="Styled__NavbarBrand-fbosc2-5 jyrqfl"
    >
      <div
        class="Styled__NavbarBurger-fbosc2-6 jrYnFG"
        data-testid="navbar-burger"
      >
        <span />
        <span />
        <span />
      </div>
    </div>
    <div
      class="Styled__NavbarMenu-fbosc2-2 gYCmeq"
    >
      <div
        class="Styled__NavbarStart-fbosc2-3 llRien"
      >
        <a
          class="Styled__NavbarItemLink-fbosc2-9 lnJXgk"
          href="/"
        >
          Home
        </a>
        <a
          class="Styled__NavbarItemLink-fbosc2-9 lnJXgk"
          href="/appointments"
        >
          Appointments
        </a>
        <a
          class="Styled__NavbarItemLink-fbosc2-9 lnJXgk"
          href="/waitingroom"
        >
          Waiting Room
        </a>
        <a
          class="Styled__NavbarItemLink-fbosc2-9 lnJXgk"
          href="/patients"
        >
          Patients
        </a>
        <div
          class="Styled__NavbarItemDiv-fbosc2-10 dvyfkx"
        >
          <span
            class="Styled__NavbarLabel-fbosc2-0 qZZfM"
          >
            Accounts
          </span>
          <div
            class="Styled__NavbarDropDown-fbosc2-7 dMqvKL"
          >
            <a
              class="Styled__NavbarItemLink-fbosc2-9 lnJXgk"
              href="/sales"
            >
              Sales
            </a>
            <a
              class="Styled__NavbarItemLink-fbosc2-9 lnJXgk"
              href="/expenses"
            >
              Expenses
            </a>
            <a
              class="Styled__NavbarItemLink-fbosc2-9 lnJXgk"
              href="/distributions"
            >
              Distributions
            </a>
            <a
              class="Styled__NavbarItemLink-fbosc2-9 lnJXgk"
              href="/journal"
            >
              Journal
            </a>
            <a
              class="Styled__NavbarItemLink-fbosc2-9 lnJXgk"
              href="/reconciliation"
            >
              Reconciliation
            </a>
            <a
              class="Styled__NavbarItemLink-fbosc2-9 lnJXgk"
              href="/acir"
            >
              AIR - Claiming
            </a>
            <a
              class="Styled__NavbarItemLink-fbosc2-9 lnJXgk"
              href="/claiming"
            >
              Claiming
            </a>
          </div>
        </div>
        <div
          class="Styled__NavbarItemDiv-fbosc2-10 dvyfkx"
        >
          <span
            class="Styled__NavbarLabel-fbosc2-0 qZZfM"
          >
            More
          </span>
          <div
            class="Styled__NavbarDropDown-fbosc2-7 dMqvKL"
          >
            <a
              class="Styled__NavbarItemLink-fbosc2-9 lnJXgk"
              href="/resources"
            >
              Resources
            </a>
            <a
              class="Styled__NavbarItemLink-fbosc2-9 lnJXgk"
              href="/settings"
            >
              Settings
            </a>
            <a
              class="Styled__NavbarItemLink-fbosc2-9 lnJXgk"
              href="/reporting"
            >
              Reporting
            </a>
            <a
              class="Styled__NavbarItemLink-fbosc2-9 lnJXgk"
              href="/audit"
            >
              Audit
            </a>
            <a
              class="Styled__NavbarItemLink-fbosc2-9 lnJXgk"
              href="/utilities"
            >
              Utilities
            </a>
          </div>
        </div>
      </div>
      <div
        class="Styled__NavbarEnd-fbosc2-4 inqMzr"
      >
        <div
          class="Styled__NavbarItemDiv-fbosc2-10 dvyfkx"
        >
          <span
            class="Styled__NavbarLabel-fbosc2-0 qZZfM"
          >
            <span
              class="Icon__User2-s196jlqe-1 ebmrAE"
            />
          </span>
          <div
            class="Styled__NavbarDropDown-fbosc2-7 dJiMnP"
          >
            <div
              class="Styled__NavbarItem-fbosc2-8 kEPPMF"
            >
              Logout
            </div>
          </div>
        </div>
      </div>
    </div>
  </nav>
</div>
`;
